mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -2107,14 +2108,46 @@ static FailureOr<rock::TransformMapAttr> removeUpperDimsFromMap(
           origLowerBounds[lowerDim] = origUpperBounds[upperDim];
         }
         for (auto [dim, subDimInfo] : removedSubDims) {
+          LLVM_DEBUG(llvm::dbgs() << "copying removedSubDimInfo from:" << dim
+                                  << " to:" << upperToLower[dim] << "\n");
           newRemovedSubDims[upperToLower[dim]] = subDimInfo;
         }
-        [[fallthrough]];
+        llvm::copy(tr.getParams(), std::back_inserter(args.params));
+        break;
       }
-      default: {
+      case TransformType::Pad: {
+        for (auto [idx, upperDim] : llvm::enumerate(tr.getUpperDims())) {
+          LLVM_DEBUG(llvm::dbgs() << "pad upper dim = " << upperDim << "\n");
+          if (removedSubDims.contains(upperDim)) {
+            // If the padded dimension is being sub-tiled, then the padding is
+            // meaningless because it happens at the either boundary and only
+            // some tiles will have the effect. For all GPU usecases, we would
+            // materialize the padded region in sub tiles. Thus, we ignore the
+            // padding in sub-tile views.
+            LLVM_DEBUG(
+                llvm::dbgs()
+                << "Some parts of the padded dimension is to be removed.\n");
+            const auto lowerDim = tr.getLowerDims()[idx];
+            origLowerBounds[lowerDim] = origUpperBounds[upperDim];
+            args.params.append({0, 0});
+          } else {
+            args.params.append(
+                {tr.getParams()[idx * 2], tr.getParams()[idx * 2 + 1]});
+          }
+        }
+        break;
+      }
+      case TransformType::Broadcast:
+      case TransformType::AddDim:
+      case TransformType::ConstDim: {
         llvm::copy(tr.getParams(), std::back_inserter(args.params));
         break;
       }
+      default: {
+        LLVM_DEBUG(llvm::dbgs()
```

Comment:
llvm::errs() ?

Reply:
hmmm we dont know whether this is an error... yet. It depends on the caller. Caller could also test the function and handle if it fails.